pkg/scheduler/plugins/proportion/proportion.go

```diff
@@ -124,17 +124,20 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 				continue
 			}
 
-			attr.deserved.Add(remaining.Clone().Multi(float64(attr.weight) / float64(totalWeight)))
+			increasedDeserved := remaining.Clone().Multi(float64(attr.weight) / float64(totalWeight))
+			oldDeserved := attr.deserved.Clone()
+			attr.deserved.Add(increasedDeserved)
 			if !attr.deserved.LessEqual(attr.request) {
 				attr.deserved = helpers.Min(attr.deserved, attr.request)
 				meet[attr.queueID] = struct{}{}
+				increasedDeserved = attr.deserved.Clone().Sub(oldDeserved)
 			}
 			pp.updateShare(attr)
 
 			glog.V(4).Infof("The attributes of queue <%s> in proportion: deserved <%v>, allocate <%v>, request <%v>, share <%0.2f>",
 				attr.name, attr.deserved, attr.allocated, attr.request, attr.share)
 
-			deserved.Add(attr.deserved)
```

Comment:
Changed to: deserved.Add(attr.deserved.Clone().sub(oldDeserved))

Reply:
Updated